src/ffi.rs

```diff
@@ -453,77 +453,149 @@ impl GGeom {
         result
     }
 
-    pub fn is_ring(&self) -> bool {
+    pub fn is_ring(&self) -> GeosResult<bool> {
         let rv = unsafe { GEOSisRing(self.c_obj()) };
-        return if rv == 1 { true } else { false };
+        if rv == 1 {
```

Comment:
since the pattern can be found several times, maybe we can wrap this in a method:


```rust
fn check_geos_predicate<F>(val: i32, on_error: F) -> GeosResult<bool>
where
    F: FnOnce() -> Error {
    match val {
        1 => Ok(true),
        0 => Ok(false),
        _ => Err(on_error())
    } 
}
```

I'm not sure about the error function, but this way the error (and the string) is created only on error ([clippy](https://github.com/rust-lang-nursery/rust-clippy) prefer `Option::ok_or_else` to `Option::ok_or` for this reason)

Reply:
Sure I guess it will be really more clean. I'm not sure about the error function either, especially if we want to keep the name of the function that triggers the error.

Reply:
I made a `check_geos_predicate` function which take the return value and an EnumVariant. This way I guess the error and the string are only created on error as you suggest (but I guess it's not really elegant to match on this new big Enum  in order to create the error), so don't hesitate to suggest me a better idea!

Reply:
you could just have kept the error as before no ?
```rust
    pub fn is_ring(&self) -> GeosResult<bool> {
        let rv = unsafe { GEOSisRing(self.c_obj()) };
        check_geos_predicate(rv, || Error::GeosError("computing isRing".into()))
    }
```

but your solution is interesting as it removes the string.

It may be overkill, but maybe we could leverage the [Failure](https://github.com/rust-lang-nursery/failure) crate for this ?

I pushed a [toy implementation](https://github.com/antoine-de/rust-geos/commit/ce9794510a6ff8fb1cee7ece25cd28e9eeba3fed)

honestly I'm not convinced that the complexity is worth it compared to the initial version with a lambda for the error. What do you think ?

I added some unit test that could be nice to get though.

Reply:
Excepting this big PredicateType Enum, I find your solution pretty elegant, so I changed the code to follow it. Sorry for the lack of imagination of mine to tackle this issue.

This is a detail but as we are referring to 'libgeos method' in the error message, i don't think it really makes sense to transform the Enum member to snake case as they match GEOS function names in camel case (?)

Reply:
ok, it's up to you, both are fine for me